Add  some WEEDO  printers

WinstonMao opened the pull request.

x40-Community commented:
As with Creality, the inherits from the base is big crap. Printers defined in the material files are then not recognized and temperature deviations in the printer profile, for example, have to be corrected.
I don't like it!

WinstonMao commented:
> As with Creality, the inherits from the base is big crap. Printers defined in the material files are then not recognized and temperature deviations in the printer profile, for example, have to be corrected. I don't like it!
> 
> The printing profiles for the generic materials are missing.

Sorry about that. We will fix the temperature setting errors and update the PR.

nallath commented on resources/definitions/weedo_base.def.json:
```diff
@@ -0,0 +1,448 @@
+      "value": "round(speed_print * 0.65,1)"
+    },
+    "speed_support": {
+      "value": "round(speed_print*0.82,1)"
```
```suggestion
      "value": "round(speed_print * 0.82, 1)"
```

nallath commented on resources/definitions/weedo_base.def.json:
```diff
@@ -0,0 +1,448 @@
+      "value": "speed_support"
+    },
+    "speed_support_interface": {
+      "value": "round(speed_support*0.689,1)"
```
```suggestion
      "value": "round(speed_support * 0.689, 1)"
```

nallath commented on resources/definitions/weedo_base.def.json:
```diff
@@ -0,0 +1,448 @@
+      "default_value": 60.0
+    },
+    "support_pattern": {
+      "default_value": "zigzag"
```
That is already the default value defined in fdmprinter. Try to adhere to the DRY (Do not Repeat Yourself) principle as much as possible. This will make maintaining the profile a lot easier

WinstonMao replied:
Thanks for the reviews. I will update the profiles according to all suggestions and commit again.

nallath commented:
> As with Creality, the inherits from the base is big crap. Printers defined in the material files are then not recognized and temperature deviations in the printer profile, for example, have to be corrected.
I don't like it!

Why is it crap? I don't understand what your issue is here.

x40-Community commented:
> > As with Creality, the inherits from the base is big crap. Printers defined in the material files are then not recognized and temperature deviations in the printer profile, for example, have to be corrected.
> > I don't like it!
> 
> Why is it crap? I don't understand what your issue is here.



    As with Creality, the inherits from the base is big crap. Printers defined in the material files are then not recognized and temperature deviations in the printer profile, for example, have to be corrected.
    I don't like it!

Why is it crap? I don't understand what your issue is here.

If you include a basis in the printer profile, the temperatures that were entered specifically for the printer in the material file are not read. The default values are always used.

We have a large archive with many adjusted and tested printing profiles. These can easily be adapted to the Weedo ME40, without a base file.

Example:
ASA material should have a print bed temperature of 100°C. But the Weedo X40 can only handle 80°C. In the material file, the standard value for all printers is therefore 100°C and 80°C especially for the Weedo X40.
If you use the Weedo ME40 with a base file, a bed temperature of 100°C is always read, even if you define a bed temperature of 80°C specifically for the printer.
The bug has been there for a long time...

jellespijker commented:
Can you sync your fork, with our `main`. We have added a **printer-linter** tool for our printer definitions and profiles. This should help us with processing our PR's. It performs some standard diagnostic checks and apply auto-formatting on the changed files in this PR, but you can also run it locally. See https://github.com/Ultimaker/Cura/tree/main/printer-linter

If it finds issues for which it knows a fix it should post a comment with that suggested fix such that you can easily commit that fix if you're okay with it.

WinstonMao commented:
> Can you sync your fork, with our `main`. We have added a **printer-linter** tool for our printer definitions and profiles. This should help us with processing our PR's. It performs some standard diagnostic checks and apply auto-formatting on the changed files in this PR, but you can also run it locally. See https://github.com/Ultimaker/Cura/tree/main/printer-linter
> 
> If it finds issues for which it knows a fix it should post a comment with that suggested fix such that you can easily commit that fix if you're okay with it. But please use commonsense with the suggestions because the tool is still in the early stages of development


> Can you sync your fork, with our `main`. We have added a **printer-linter** tool for our printer definitions and profiles. This should help us with processing our PR's. It performs some standard diagnostic checks and apply auto-formatting on the changed files in this PR, but you can also run it locally. See https://github.com/Ultimaker/Cura/tree/main/printer-linter
> 
> If it finds issues for which it knows a fix it should post a comment with that suggested fix such that you can easily commit that fix if you're okay with it. But please use commonsense with the suggestions because the tool is still in the early stages of development

Yes,i will do it

nallath commented:
Closing & opening it to trigger the tests again (something went wrong there)

saumyaj3 commented:
This PR has some conflicts as I already see  weedo_base.def.json in main. please allign them ?

WinstonMao commented:
> This PR has some conflicts as I already see weedo_base.def.json in main. please allign them ?

jellespijker commented on resources/extruders/weedo_x40v3_extruder_left.def.json:
```diff
@@ -0,0 +1,21 @@
+    },
+    "overrides":
+    {
+        "extruder_nr":
```
it seems the check is incorrect, please ignore this

jellespijker commented on resources/extruders/weedo_x40v3_extruder_right.def.json:
```diff
@@ -0,0 +1,20 @@
+    },
+
+    "overrides": {
+        "extruder_nr": {
```
ignore this suggestion
